static/app/views/performance/newTraceDetails/traceActionsMenu.tsx

```diff
@@ -0,0 +1,104 @@
+import {useMemo} from 'react';
+
+import {openModal} from 'sentry/actionCreators/modal';
+import {Button} from 'sentry/components/button';
+import {DropdownMenu} from 'sentry/components/dropdownMenu';
+import {IconEllipsis} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import type {EventTransaction} from 'sentry/types/event';
+import {trackAnalytics} from 'sentry/utils/analytics';
+import type EventView from 'sentry/utils/discover/eventView';
+import {SavedQueryDatasets} from 'sentry/utils/discover/types';
+import type {UseApiQueryResult} from 'sentry/utils/queryClient';
+import type RequestError from 'sentry/utils/requestError/requestError';
+import {useNavigate} from 'sentry/utils/useNavigate';
+import useOrganization from 'sentry/utils/useOrganization';
+import useProjects from 'sentry/utils/useProjects';
+import {hasDatasetSelector} from 'sentry/views/dashboards/utils';
+
+import {traceAnalytics} from './traceAnalytics';
+import {getCustomInstrumentationLink} from './traceConfigurations';
+import {TraceShortcutsModal} from './traceShortcutsModal';
+import {useHasTraceNewUi} from './useHasTraceNewUi';
+
+function TraceActionsMenu({
+  rootEventResults,
+  traceEventView,
+}: {
+  rootEventResults: UseApiQueryResult<EventTransaction, RequestError>;
+  traceEventView: EventView;
+}) {
+  const hasTraceNewUi = useHasTraceNewUi();
+  const organization = useOrganization();
+  const {projects} = useProjects();
+  const navigate = useNavigate();
+
+  const traceProject = useMemo(() => {
+    return rootEventResults.data
+      ? projects.find(p => p.id === rootEventResults.data.projectID)
+      : undefined;
+  }, [projects, rootEventResults.data]);
```

Comment:
You should avoid memoization in cases like these, memoizing has slight memory overhead and should only be used when you're trying to prevent excess re-renders or fix performance problems. This is basically an O(1) operation since it's bounded by how many projects exist in the current org, no need to memoize

Reply:
You could also place it after the `if(!hasTraceNewUi)` block since it will never be needed before that, but it doesn't matter much

Reply:
Hey yup I agree with ya on the memoization.

Reply:
I'll get rid of the `useMemo`